sympy/polys/ring_series.py

```diff
@@ -1140,6 +1140,8 @@ def rs_exp(p, x, prec):
     n = 1
     k = 1
     c = []
+    if not isinstance(prec, int):
+        prec = int(prec.p/prec.q) + 1
     for k in range(prec):
```

Comment:
would `ceiling(prec)` work here?

Reply:
Yes, `ceiling(prec)` does work here. Difference being, in the previous case `prec` comes out to be of class `int` but now it would be of class `sympy.core.numbers.Integers`.
Should I commit this change?

Reply:
I guess, that would work okay.